tests/test_messageexchange.py

```diff
@@ -0,0 +1,88 @@
+"""Test the MessageExchangeProtocol Handler in agent."""
+from aries_staticagent import StaticConnection
+import pytest
+
+
+@pytest.mark.asyncio
+async def test_first_message(connection: StaticConnection):
+    reply = await connection.send_and_await_reply_async(
+        {
+            "@type": "https://example.com/message-exchange/0.1/message1",
+            "item": "Hi! I'd like to learn more about blockchain.",
+        },
+        return_route="all",
+    )
+    assert (
+        reply["content"] == "Blockchain is a type of database that stores information "
+        "in a decentralized, distributed digital ledger."
+    )
+
+
+@pytest.mark.asyncio
+async def test_second_message(connection: StaticConnection):
+    reply = await connection.send_and_await_reply_async(
+        {
+            "@type": "https://example.com/message-exchange/0.1/message2",
+            "item": "Why is this a good way of transferring information?",
+        },
+        return_route="all",
+    )
+    assert (
+        reply["content"] == "Data is recorded in a way that is transparent, time "
+        "stamped, and immutable. (Type one of the previous words to learn more.)"
+    )
+
+
+@pytest.mark.asyncio
+async def test_third_message(connection: StaticConnection):
```

Comment:
I recommend splitting this test into three: `test_third_message_immutable`, `test_third_message_timestamped` and `test_third_message_transparent`. This will test each of the possible paths independently.